[문자열 덧셈 계산기] 김대민 미션 제출합니다.

ooortcloud opened the pull request.

ooortcloud committed:
docs(README.md): 구현할 기능 목록 정리

아래 세 가지 주제에 대해 목록을 정리했다.
- 구현: 이번 프로젝트에서 내가 구현해야 할 기능들이 무엇인지 리스트화
- 코딩 컨벤션: 이번 프로젝트 동안 따라야 하는 코딩 컨벤션이 무엇인지 리스트화
- 비즈니스: 프로젝트 외적으로 어떻게 업무를 수행해야 하는지 리스트화

ooortcloud committed:
feat: 문자열 파싱 및 연산 기능 구현

문자열 파싱
- String 객체의 split() 메소드를 사용하여 다양한 구분자(디폴트: 쉼표, 콜론)들에 대해 문자열 내 숫자들만 따로 분리함.
- 커스텀 구분자를 고려하기 위해 split() 메소드에 넣을 인자를 동적으로 처리함.
- 연속으로 구분자가 입력되는 경우를 고려함.
- 연속 구분자 입력, 커스텀 구분자 연속 입력, 2자리 이상의 숫자 입력에 대한 테스트 진행 완료.

연산
- Java Stream API를 사용하여 숫자 문자열 배열의 합을 연산.
- 각각의 테스트 케이스에 대해 연산 결과가 정상적으로 나오는 것을 확인
  완료.

ooortcloud committed:
feat(exception): 주어진 조건 외의 문자열 입력 시 예외 처리 기능 구현

사용자가 0 이하의 정수를 입력하거나 숫자가 아닌 문자를 입력하는 경우, 프로그램이 `IllegalArgumentException` 예외를 발생시키도록 함.

seong-jin-jo commented on README.md:
```diff
@@ -1 +1,31 @@
+- 제출 시작일시: 24-10-20 15:00
+- 지원 플랫폼에 로그인한 뒤 '제출하기'를 눌러 과제 제출하여 예제 테스트 실행해볼 것
+- 제출 이후에도 push를 통해 수정 가능
+- 단, 마감 이후에는 추가 push를 허용하지 않음
```
안녕하세요 대민님 코드리뷰하러 왔습니다~ 
테스트 코드를 상세히 작성하셔서 매우 안정적인 동작을 구현하신 점과 stream 을 사용하여 간결하게 코드를 작성하신 점이 인상적입니다.

seong-jin-jo commented on src/main/java/calculator/Application.java:
```diff
@@ -1,7 +1,71 @@
+
+        String delimiters = ",:";  // 커스텀 문자열까지 받기 위해 열어둔다.
+
+        Scanner sc = new Scanner(System.in);
+        String inputStr = sc.nextLine();
+        sc.close();
```
과제 안내 최하단 프로그래밍 요구 사항에 `camp.nextstep.edu.missionutils` 라이브러리에서 제공하는 Console API를 사용하도록 장려하고 있는데 이부분 대체하시면 완벽할 것 같습니다!

seong-jin-jo reviewed (commented):
LGTM

ooortcloud committed:
fix(exception): 0을 예외처리하는 방식 변경

각 단위 별 문자열 전체애 대해 그것이 0인지 아닌지 판단하는 방식으로 변경함.

그 외) 입력받는 방식을 camp.nextstep.edu.missionutils.Console을 사용하도록 변경.

ooortcloud committed:
refactor(overflow): 오버플로우 범위를 넘어서는 수에 대한 합연산 기능

BigInteger를 사용하여 오버플로우로부터 안전한 연산이 가능하도록 보장함.
